module/optee/mbx/test/mod_optee_mbx_unit_test.c

```diff
@@ -27,10 +30,369 @@ void tearDown(void)
 {
 }
 
+struct mbx_device_ctx fake_optee_mbx_ctx[FAKE_SCMI_SERVICE_IDX_COUNT];
+
+/*!
+ * \brief test module's init function.
+ *
+ *  \details Test all possible cases for init.
+ */
+void test_optee_mbx_init(void)
+{
+    int status;
+    fwk_id_t init_id;
+
+    init_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+
+    /* Element count can't be null */
+    status = mbx_init(init_id, 0, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Fail to allocate module cxt */
+    fwk_mm_calloc_ExpectAndReturn(
+        1,
+        sizeof(struct mbx_device_ctx),
+        NULL);
+    status = mbx_init(init_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_NOMEM, status);
+
+    /* Allocate module context */
+    fwk_mm_calloc_ExpectAndReturn(
+        FAKE_SCMI_SERVICE_IDX_COUNT,
+        sizeof(struct mbx_device_ctx),
+        fake_optee_mbx_ctx);
+    status = mbx_init(init_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(mbx_ctx.device_ctx_table, fake_optee_mbx_ctx);
+}
+
+/*!
+ * \brief test module's element_init function.
+ *
+ *  \details Test all possible cases for element_init.
```

Comment:
That's a generic description: we will test all possible scenarios that are supported by the function: In this case it's only 1 test because there is on additional check done during init as the module trust fwk  and scp core aprt